epytope/Core/Result.py

```diff
@@ -110,20 +130,33 @@ def merge_results(self, others):
 
         if type(others) == type(self):
             others = [others]
+        
+        # Concatenates self and to be merged dataframe(s)
+        for other in others:
+            df = pandas.concat([df, other], axis=1)
 
-        for i in range(len(others)):
-            df1a, df2a = df.align(others[i])
-            zero1 = df1a == 0
-            zero2 = df2a == 0
-            df1a = df1a.fillna(0)
-            df2a = df2a.fillna(0)
-            df = df1a+df2a
-            true_zero = zero1 | zero2
-            false_zero = df == 0
-            zero = true_zero & false_zero
-            nans = ~true_zero & false_zero
-            df[zero] = 0
-            df[nans] = numpy.NaN
+        # Merge result of multiple predictors in others per allele
+        df_merged = pandas.concat([group[1] for group in df.groupby(level=[0,1], axis=1)], axis=1)
+    
+        return EpitopePredictionResult(df_merged)
+
+    def from_dict(d, peps, method):
```

Comment:
Would it make sense to use the Pandas function `from_dict` here?

Reply:
Unfortunately not.. I tested that at the start of the implementation, but it is to complicated with the 3-layer multi index